content/en/blog/_posts/2021-08-18-alpha-swap-support.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: blog
+title: 'New in Kubernetes v1.22: alpha support for using swap memory'
+date: 2021-08-18
+slug: run-nodes-with-swap-alpha
+---
+
+**Author:** Elana Hashman (Red Hat)
+
+The 1.22 release introduced alpha support for configuring swap memory usage for
+Kubernetes workloads on a per-node basis.
+
+In prior releases, Kubernetes did not support the use of swap memory on Linux,
+as it is difficult to provide guarantees and account for pod memory utilization
+when swap is involved. As part of Kubernetes' earlier design, swap support was
+considered out of scope, and a kubelet would by default fail to start if swap
+was detected on a node.
+
+However, there are a number of [use cases](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#user-stories)
+that would benefit from Kubernetes nodes supporting swap, including improved
+node stability, better support for applications with high memory overhead but
+smaller working sets, the use of memory-constrained devices, and memory
+flexibility.
+
+Hence, over the past two releases, [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node#readme) has
+been working to gather appropriate use cases and feedback, and propose a design
+for adding swap support to nodes in a controlled, predictable manner so that
+Kubernetes users can perform testing and provide data to continue building
+cluster capabilities on top of swap. The alpha graduation of swap memory
+support for nodes is our first milestone towards this goal!
+
+## How does it work?
+
+There are a number of possible ways that one could envision swap use on a node.
+To keep the scope manageable for this initial implementation, when swap is
+already provisioned and available on a node, [we have proposed](https://github.com/kubernetes/enhancements/blob/9d127347773ad19894ca488ee04f1cd3af5774fc/keps/sig-node/2400-node-swap/README.md#proposal)
+the kubelet should be able to be configured such that:
+
+- It can start with swap on.
+- It will direct the CRI to allocate Kubernetes workloads zero swap memory by
```

Comment:
This sentence reads a little weird. Maybe:

```
It will direct the Container Runtime Interface to allocate zero swap memory to Kubernetes workloads by default.
```

or similar.

Reply:
Done.